### PR TITLE
Fix payload mapping for multiple targets

### PR DESCRIPTION
**Pull request by dguggemos**
If multiple targets of a connection defined different payload mappings, only the first mapping was applied.

### CODE REVIEW

File: services/models/connectivity/src/main/java/org/eclipse/ditto/services/models/connectivity/MappableOutboundSignal.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2017 Contributors to the Eclipse Foundation
```
Comment (thjaeckle) [inline services/models/connectivity/src/main/java/org/eclipse/ditto/services/models/connectivity/MappableOutboundSignal.java]:
```suggestion
Copyright (c) 2019 Contributors to the Eclipse Foundation
```

File: services/models/connectivity/src/main/java/org/eclipse/ditto/services/models/connectivity/MappedOutboundSignal.java [hunk 2]
```diff
@@ -68,27 +68,24 @@ public JsonObject toJson(final JsonSchemaVersion schemaVersion, final Predicate<
...
     @Override
     public boolean equals(final Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (!(o instanceof MappedOutboundSignal)) {
-            return false;
-        }
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (jufickel-b) [inline services/models/connectivity/src/main/java/org/eclipse/ditto/services/models/connectivity/MappedOutboundSignal.java]: The parentheses got lost somehow.

File: services/models/connectivity/src/main/java/org/eclipse/ditto/services/models/connectivity/OutboundSignal.java [hunk 3]
```diff
@@ -60,6 +61,18 @@ default JsonObject toJson(final JsonSchemaVersion schemaVersion, final JsonField
         return toJson(schemaVersion, FieldType.notHidden()).get(fieldSelector);
     }
 
+    /**
+     * Extends the {@link OutboundSignal} by adding the payload mapping used to map the signal.
+     */
+    interface Mappable extends OutboundSignal {
```
Comment (jufickel-b) [inline services/models/connectivity/src/main/java/org/eclipse/ditto/services/models/connectivity/OutboundSignal.java]: We should move the Mappable interface to its own file and rename it to MappableOutboundSignal. Nesting it in its super-type confused me at first.

File: services/models/connectivity/src/test/java/org/eclipse/ditto/services/models/connectivity/MappableOutboundSignalTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.eclipse.ditto.model.connectivity.PayloadMapping;
+import org.junit.Test;
+import org.mutabilitydetector.unittesting.AllowedReason;
+import org.mutabilitydetector.unittesting.MutabilityAssert;
+import org.mutabilitydetector.unittesting.MutabilityMatchers;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+public class MappableOutboundSignalTest {
```
Comment (jufickel-b) [inline services/models/connectivity/src/test/java/org/eclipse/ditto/services/models/connectivity/MappableOutboundSignalTest.java]: Is it intended that this class can be sub-classed? If so, it should be abstract or else final.

File: services/models/connectivity/src/test/java/org/eclipse/ditto/services/models/connectivity/MappableOutboundSignalTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                AllowedReason.provided(OutboundSignal.class, PayloadMapping.class).areAlsoImmutable());
+    }
+
+    @Test
+    public void testHashCodeAndEquals() {
+        EqualsVerifier.forClass(MappableOutboundSignal.class)
+                .usingGetClass()
+                .verify();
+    }
+
```
Comment (jufickel-b) [inline services/models/connectivity/src/test/java/org/eclipse/ditto/services/models/connectivity/MappableOutboundSignalTest.java]: I suggest to add a test case for getting the payload mapping as this is the core functionality of the tested class.

**Pull request merged**